pkg/schemacmp/lattice_test.go

```diff
@@ -0,0 +1,330 @@
+		},
+		{
+			a:            Singleton(123),
+			b:            Singleton(2468),
+			compareError: `distinct singletons.*`,
+			joinError:    `distinct singletons.*`,
+		},
+		{
+			a:            BitSet(0b010110),
+			b:            BitSet(0b110001),
+			compareError: `non-inclusive bit sets.*`,
+			join:         BitSet(0b110111),
+		},
+		{
+			a:             BitSet(0xffffffff),
+			b:             BitSet(0),
+			compareResult: 1,
+			join:          BitSet(0xffffffff),
+		},
+		{
+			a:             BitSet(0b10001),
+			b:             BitSet(0b11011),
+			compareResult: -1,
+			join:          BitSet(0b11011),
+		},
+		{
+			a:             BitSet(0x522),
+			b:             BitSet(0x522),
+			compareResult: 0,
+			join:          BitSet(0x522),
+		},
+		{
+			a:             Byte(123),
+			b:             Byte(123),
+			compareResult: 0,
+			join:          Byte(123),
+		},
+		{
+			a:             Byte(1),
+			b:             Byte(23),
+			compareResult: -1,
+			join:          Byte(23),
+		},
+		{
+			a:             Byte(123),
+			b:             Byte(45),
+			compareResult: 1,
+			join:          Byte(123),
+		},
+		{
+			a:            Tuple{Byte(123), Bool(false)},
+			b:            Tuple{Byte(67), Bool(true)},
+			compareError: `at tuple index 1: combining contradicting orders.*`,
+			join:         Tuple{Byte(123), Bool(true)},
+		},
+		{
+			a:             Tuple{},
+			b:             Tuple{},
+			compareResult: 0,
+			join:          Tuple{},
+		},
+		{
+			a:            Tuple{Singleton(6), Singleton(7)},
+			b:            Tuple{Singleton(6), Singleton(8)},
+			compareError: `at tuple index 1: distinct singletons.*`,
+			joinError:    `at tuple index 1: distinct singletons.*`,
+		},
+		{
+			a:            Tuple{},
+			b:            Tuple{Bool(false)},
+			compareError: `tuple length mismatch.*`,
+			joinError:    `tuple length mismatch.*`,
+		},
+		{
+			a:            Bool(false),
+			b:            Singleton(false),
+			compareError: `type mismatch.*`,
+			joinError:    `type mismatch.*`,
+		},
+		{
+			a:            Maybe(Singleton(123)),
+			b:            Maybe(Singleton(678)),
+			compareError: `distinct singletons.*`,
+			joinError:    `distinct singletons.*`,
+		},
+		{
+			a:             Maybe(Byte(111)),
+			b:             Maybe(Byte(222)),
+			compareResult: -1,
+			join:          Maybe(Byte(222)),
+		},
+		{
+			a:             Maybe(nil),
+			b:             Maybe(Singleton(135)),
+			compareResult: -1,
+			join:          Maybe(Singleton(135)),
+		},
+		{
+			a:             Maybe(nil),
+			b:             Maybe(nil),
+			compareResult: 0,
+			join:          Maybe(nil),
+		},
+		{
+			a:            Bool(false),
+			b:            Maybe(Bool(false)),
+			compareError: `type mismatch.*`,
+			joinError:    `type mismatch.*`,
+		},
+		{
+			a:             StringList{"one", "two", "three"},
+			b:             StringList{"one", "two", "three", "four", "five"},
+			compareResult: -1,
+			join:          StringList{"one", "two", "three", "four", "five"},
```

Comment:
OK, but the result should be "Incompatible", since a `StringList` only compare the start of the list.

This is because `ENUM('two', 'three')` (two = 0, three = 1) is incompatible with `ENUM('one', 'two', 'three')` (one = 0, two = 1, three = 2), and similarly `SET('two', 'three')` (two = 1, three = 2) is incompatible with `SET('one', 'two', 'three')` (one = 1, two = 2, three = 4).